sources/web/datalab/jupyter.ts

```diff
@@ -64,14 +66,19 @@ var templates: common.Map<string> = {
  */
 var appSettings: common.Settings;
 
-function pipeOutput(stream: NodeJS.ReadableStream, port: number, error: boolean) {
+function pipeOutput(server: JupyterServer, stream: NodeJS.ReadableStream, port: number, error: boolean) {
   stream.setEncoding('utf8');
+  var listeningMessage = `The Jupyter Notebook is running at: http://127.0.0.1:${port}`;
+
   stream.on('data', (data: string) => {
     // Jupyter generates a polling kernel message once every 3 seconds
     // per kernel! This adds too much noise into the log, so avoid
     // logging it.
 
     if (data.indexOf('Polling kernel') < 0) {
+      if (data.indexOf(listeningMessage) >= 0) {
```

Comment:
> is it possible the message appears before we start listening to it?

Ah, there is a risk here... the message could be output to the stderr before we start reading the (streamed) stderr.

OK; this won't work reliably.

I'll remove this check for now and try to come up with a more reliable one later.